### PR TITLE
#119 #120 Fix make_patch() to avoid casting numeric string to int on item moved

### PR DESCRIPTION
**Pull request by RyanSept**
Addresses #119 #120

### CODE REVIEW

**Comment by stefankoegl** [comment]
There seems to be an issue with a pypy build, but that seems unrelated to your pull request.

**Comment by RyanSept** [comment]
Thanks 🙌

File: jsonpatch.py [hunk 1]
```diff
@@ -800,7 +802,8 @@ def _item_removed(self, path, key, item):
         new_index = self.insert(new_op)
         if index is not None:
             op = index[2]
-            if type(op.key) == int:
+            added_item = op.pointer.to_last(self.dst_doc)[0]
+            if type(added_item) == list:
```
Comment (stefankoegl) [inline jsonpatch.py:806]: Could you please just add a comment here to clarify for posterity?

**Review by stefankoegl — CHANGES_REQUESTED** [review]

File: jsonpatch.py (hunk 1 above)
Comment (RyanSept) [reply jsonpatch.py]: I've done so now. Please let me know if there's anything else I should add in the comment. It was a bit difficult to express what the _on_undo_add call accomplishes and I think it'd be valuable to mention that.

Comment (RyanSept) [reply jsonpatch.py]: @stefankoegl kind reminder on this

**Comment by stefankoegl** [comment]
Thanks :)

**Pull request merged**